File: year2/semestrul2/MPP/Laboratoare/AtletismFinal/Repository/ParticipantRepository.cs
using AtletismFinal.Domain;
using System.Collections.Generic;

namespace AtletismFinal.Repository
{
    public interface ParticipantRepository
    {
        List<Participant> FindAll();

        List<Participant> FindByNameAndSurname(string name, string surname);

        List<Participant> FindByEmail(string email);
    }
}

File: year2/semestrul2/MPP/Laboratoare/lab4recuperare/src/main/java/org/example/Repository/Interfaces/ParticipantRepository.java
package org.example.Repository.Interfaces;

import org.example.Model.Participant;

import java.util.List;

public interface ParticipantRepository {
    List<Participant> findAll();

    List<Participant> findByNameandSurname(String name, String surname);

    List<Participant> findByEmail(String email);
}
